test/autorest/lrogroup/lros_test.go

```diff
@@ -37,22 +39,23 @@ func httpClientWithCookieJar() azcore.Transport {
 	})
 }
 
-// func TestLROResumeWrongPoller(t *testing.T) {
-// 	op := getLROSOperations(t)
-// 	resp, err := op.Delete202NoRetry204(context.Background())
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	poller := resp.GetPoller()
-// 	rt, err := poller.ResumeToken()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	_, err = op.ResumeProductPoller(rt)
-// 	if err == nil {
-// 		t.Fatal("expected an error but did not find receive one")
-// 	}
-// }
+func TestLROResumeWrongPoller(t *testing.T) {
+	// TODO discuss what sort of check we want with the new design to avoid resuming from the wrong poller type
+	op := getLROSOperations(t)
+	resp, err := op.BeginDelete202NoRetry204(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	rt, err := poller.ResumeToken()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = op.ResumePost200WithPayload(rt)
```

Comment:
Might be good to capture the poller and ensure it's nil.

Reply:
Done